Automattic/jetpack
# Jetpack: update full-screen connection banner to Emerald design

Fixes #29051

## Proposed changes:

This PR removes the legacy full-screen connection banner. Instead, we re-use the existing connection screen component currently in use in the Jetpack dashboard.

**Visuals**

| **Before** | **After** |
| ---------- | --------- |
| <img width="1385" alt="image" src="https://user-images.githubusercontent.com/426388/226416397-3c7b0c90-8dd8-4b2d-8d27-cfd2ab810a35.png"> | <img width="1388" alt="image" src="https://user-images.githubusercontent.com/426388/226433407-db062ba9-c35f-4993-ab6f-50aaaddc9bd8.png"> |
|   |  |


**To do**

- [x] Ensure that the connection is still tracked with the same parameters, to avoid breaking funnels.
- [x] Add a dismiss button so folks can choose not to connect.
    - [x] That dismiss button should be controlled by the same logic as before.
- [x] Ensure the screen looks good on all screen sizes.
- [x] ~Just like the previous screen, I am a bit worried about a11y, since it's an overlay that doesn't remove the existing elements on the page.~ I chose to wrap the component in a modal, to clearly delimitate the content that's popping up in a more accessible way.
- [ ] Ensure that none of the other banners are impacted.
    - [x] Regular connection banner appearing on all pages.
    - [x] connection screen in the Jetpack dashboard.
    - [ ] License-aware connection banner.


### Other information:


## Jetpack product discussion

See the conversation in #29051 to find out more.

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

This is a bit tricky to test, since the banner only appears once you first activate the plugin. One way to repeat test on the same site is to use the WP CLI to "reset" a site:
`wp transient set activated_jetpack true`
If you type this in after disconnecting your site from WordPress.com, you should see the full-screen banner:

- In the Plugins menu
- In the Dashboard menu

If you reload the page, that banner should be gone.

Make sure you check:

- In different browsers and devices.

*More testing instructions to follow*
